openhtf/core/test_state.py

```diff
@@ -416,22 +418,12 @@ def _finalize_measurements(self):
     for meas in self.measurements.values():
       meas.set_notification_callback(None)
 
-    # Initialize with already-validated and UNSET measurements.
-    validated_measurements = {
-        name: measurement
-        for name, measurement in self.measurements.iteritems()
-        if measurement.outcome is not measurements.Outcome.PARTIALLY_SET
-    }
-
     # Validate multi-dimensional measurements now that we have all values.
-    validated_measurements.update({
-        name: measurement.validate()
-        for name, measurement in self.measurements.iteritems()
-        if measurement.outcome is measurements.Outcome.PARTIALLY_SET
-    })
+    for measurement in self.measurements.itervalues():
+      measurement.validate()
 
     # Fill out final values for the PhaseRecord.
-    self.phase_record.measurements = validated_measurements
+    self.phase_record.measurements = self.measurements.copy()
```

Comment:
Why do we need to copy this? This is in `_finalize_measurements` which implies not touching these anymore